lua/sql/parser.lua

```diff
@@ -0,0 +1,147 @@
+local u = require'sql.utils'
+local M = {}
+
+local keynames = function(params, kv)
+  local keys = u.keys(params)
+  local s = kv and ":" or ""
+  local res = {}
+  u.map(keys, function(k)
+    res[#res+1] = s .. k
+  end)
+  return res
+end
+
+local booleansql = function(value)
+  if type(value) == "boolean" then
+    value = (value == true) and 1 or 0
+  end
+  return value
+end
+
+
+--- format values part of sql statement
+---@params defs table: key/value paris defining sqlite table keys.
+---@params defs kv: whether to bind by named keys.
+M.keys = function(defs, kv)
+  if not defs then return end
+  kv = kv == nil and true or kv
+  if kv then
```

Comment:
Thats weird. 
If key is nil then run it because we set it to true or if kv is not nil then set kv and still run the if.

Reply:
I'm trying to say, unless kv is false then do ...